File: publicq-main/client/src/models/file-upload-request.ts
export interface FileUploadRequest {
  /**
   * Module identifier to which the static file belongs.
   */
  moduleId: string; // GUID as string

  /**
   * File to be uploaded.
   */
  file?: File; // optional

  /**
   * Indicates if the file is being uploaded at module level (true) or question/answer level (false).
   */
  isModuleLevelFile?: boolean;
}

File: publicq-main/src/PublicQ.API/Models/Requests/FileUploadRequest.cs
namespace PublicQ.API.Models.Requests;

/// <summary>
/// Static file upload request model.
/// </summary>
public class FileUploadRequest
{
    /// <summary>
    /// Module identifier to which the static file belongs.
    /// </summary>
    public Guid ModuleId { get; set; }
    
    /// <summary>
    /// File to be uploaded.
    /// </summary>
    public IFormFile? File { get; set; }
    
    /// <summary>
    /// Indicates if the file is being uploaded at module level (true) or question/answer level (false).
    /// </summary>
    public bool IsModuleLevelFile { get; set; }
}